src/tests/ftest/util/server_utils.py

```diff
@@ -245,198 +322,351 @@ def __init__(self):
-        for index in range(len(self.server_params)):
-            yaml_data["servers"].append({})
-            for name in self.server_params[index].get_param_names():
-                value = getattr(self.server_params[index], name).value
-                if value is not None and value is not False:
-                    yaml_data["servers"][index][name] = value
-
-        # Write default_value_set dictionary in to AVOCADO_FILE
-        # This will be used to start with daos_server -o option.
-        try:
-            with open(filename, 'w') as write_file:
-                yaml.dump(yaml_data, write_file, default_flow_style=False)
-        except Exception as error:
-            print("<SERVER> Exception occurred: {0}".format(error))
-            raise ServerFailed(
-                "Error writing daos_server command yaml file {}: {}".format(
-                    filename, error))
-        return filename
+        super(DaosServerCommand, self).__init__(
+            "/run/daos_server/*", "daos_server", path, yaml_cfg, timeout)
+        self.pattern = self.NORMAL_PATTERN
+
+        # If specified use the configuration file from the YamlParameters object
+        default_yaml_file = None
+        if isinstance(self.yaml, YamlParameters):
+            default_yaml_file = self.yaml.filename
+
+        # Command line parameters:
+        # -d, --debug        Enable debug output
+        # -j, --json         Enable JSON output
+        # -o, --config-path= Path to agent configuration file
+        self.debug = FormattedParameter("--debug", True)
+        self.json = FormattedParameter("--json", False)
+        self.config = FormattedParameter("--config={}", default_yaml_file)
+
+        # Additional daos_server command line parameters:
+        #     --allow-proxy  Allow proxy configuration via environment
+        self.allow_proxy = FormattedParameter("--allow-proxy", False)
+
+        # Used to override the sub_command.value parameter value
+        self.sub_command_override = None
+
+    def get_sub_command_class(self):
+        """Get the daos_server sub command object based upon the sub-command."""
+        if self.sub_command_override is not None:
+            # Override the sub_command parameter
+            sub_command = self.sub_command_override
+        else:
+            # Use the sub_command parameter from the test's yaml
+            sub_command = self.sub_command.value
+
+        # Available daos_server sub-commands:
+        #   network  Perform network device scan based on fabric provider
+        #   start    Start daos_server
+        #   storage  Perform tasks related to locally-attached storage
+        if sub_command == "network":
+            self.sub_command_class = self.StartSubCommand()
+        elif sub_command == "start":
+            self.sub_command_class = self.StartSubCommand()
+        elif sub_command == "storage":
+            self.sub_command_class = self.StorageSubCommand()
```

Comment:
(pylint-redefined-variable-type) Redefinition of self.sub_command_class type from util.server_utils.DaosServerCommand.StartSubCommand to util.server_utils.DaosServerCommand.StorageSubCommand